bin/hdfcoinc/pycbc_coinc_findtrigs

```diff
@@ -239,94 +252,153 @@ for tnum in template_ids:
 
-            #index values of the zerolag triggers
+            # index values of the zerolag triggers
             fi = numpy.where(slide == 0)[0]
 
-            #index values of the background triggers
+            # index values of the background triggers
             bi = numpy.where(slide != 0)[0]
-            logging.info('%s foreground triggers' % len(fi))
-            logging.info('%s background triggers' % len(bi))
-
-            # coincs will be decimated by successive (multiplicative) levels
-            # tracked by 'total_factor'
-            bi_dec = bi.copy()
-            dec = numpy.ones(len(bi))
-
-            total_factor = 1
-            for decstr in args.loudest_keep_values:
-                thresh, factor = decstr.split(':')
-                thresh = float(thresh)
-                # throws an error if 'factor' is not the string representation
-                # of an integer
-                total_factor *= int(factor)
-
-                # triggers not being further decimated
-                upper = c[bi_dec] >= thresh
-                idxk = bi_dec[upper]
-
-                # decimate the remaining triggers
-                idx = bi_dec[c[bi_dec] < thresh]
-                idx = idx[slide[idx] % total_factor == 0]
-
-                bi_dec = numpy.concatenate([idxk, idx])
-                dec = numpy.concatenate([dec[upper],
-                                         numpy.repeat(total_factor, len(idx))]
-                                       )
-
-            ti = numpy.concatenate([bi_dec, fi]).astype(numpy.uint32)
-            dec_fac = numpy.concatenate([dec, numpy.ones(len(fi))])
-            logging.info('%s after decimation' % len(ti))
-
-            # temporary storage for decimated trigger ids
-            g0 = i0[ti]
-            g1 = i1[ti]
-            del i0
-            del i1
-
-            data['stat'] += [c[ti]]
-            data['decimation_factor'] += [dec_fac]
-            data['time1'] += [t0[g0]]
-            data['time2'] += [t1[g1]]
-            data['trigger_id1'] += [tid0[g0]]
-            data['trigger_id2'] += [tid1[g1]]
-            data['timeslide_id'] += [slide[ti]]
-            data['template_id'] += [numpy.repeat(tnum, len(ti))]
+
+            # keep foreground triggers and background triggers below
+            # the lowest decimation threshold
+            bl = bi[c[bi] < threshes[-1]]
+            ti = numpy.concatenate([fi, bl])
+
+            i0 = i0[ti]
+            i1 = i1[ti]
+            c = c[ti]
+            slide = slide[ti]
+            dec = numpy.concatenate([numpy.ones(len(fi)), numpy.repeat(total_factors[-1], len(bl))])
+
+            try:
+                s1stat = s1['snglstat'].copy()
+            except IndexError:
+                s1stat = s1.copy()
+
+            if not rank_method.single_increasing:
+                s1stat *= -1.
+
+            tidx = len(threshes)
+            for i in range(1, len(threshes)):
```

Comment:
a comment for explanation of what is happening in this loop would be good - it took me a few minutes to work out what was happening here

Reply:
Does this comment read okay?

Reply:
looks good to me